contentScript.js

```diff
@@ -0,0 +1,149 @@
+			while(oldBms[0]) { oldBms[0].parentNode.removeChild(oldBms[0]);}
+		}
+	
+		/* get current video's duration */
+		const getVideoDuration = () => {
+			if (ytPlayer) { return parseInt(ytPlayer.duration);}
+			return null;
+		}
+
+			/* Handle request to add new bookmark */
+			const addNewBmEventHandler = () => {
+	
+		let currentTime = null;
+		if (ytPlayer) {
+		 currentTime = parseInt(ytPlayer.currentTime);
+		}
+
+		if (currentTime) {
+
+			const newBm = { time: currentTime, desc: 'Bookmark at ' + getTime(currentTime) };
+
+			fetchBookmarks().then(obj => {
+				
+				BmsList = obj;
+				BmsList.push(newBm);
+				BmsList = BmsList.sort((a, b) => ( a.time - b.time));
+				chrome.storage.sync.set({[currentVid]: JSON.stringify(BmsList)});
+			});
+		}
+	}
+
+		/* this checks for whenever a new video is loaded or the current tab is reloaded */
+		const newVideoLoaded = (refreshed) => {
+
+			delOldBms();
+			BmsList = [];
+			ytLeftControls = document.getElementsByClassName('ytp-left-controls')[0];
+			ytPlayer = document.getElementsByClassName("html5-main-video")[0];
+		
+	
+			const fetchBmsPromise = fetchBookmarks();
+	
+			const videoBmTimePromise = new Promise((resolve) => {
+				const getCurrentTime = () => {
+					ytPlayer.removeEventListener('playing', getCurrentTime);
+					resolve(getVideoDuration());
+				}
+	
+				if (refreshed) {
+					resolve(getVideoDuration());
+				} else {
+					ytPlayer.addEventListener('playing', getCurrentTime);
+				}
+			});
+	
+			Promise.all([fetchBmsPromise, videoBmTimePromise])
+				.then(obj => {
+					BmsList = obj[0];
+	
+				// Adding bookmark button 
+				let bmBtn = document.getElementsByClassName('bookmark-btn')[0];
+	
+				if (!bmBtn) {
+					const bmBtn = document.createElement('img');
+		
+					bmBtn.src = chrome.runtime.getURL('assets/bookmark.png');
+					bmBtn.className = 'ytp-button ' + 'bookmark-btn';
+					bmBtn.title = 'Click to bookmark current timestamp';
+		
+					ytLeftControls.appendChild(bmBtn);
+		
+					bmBtn.addEventListener('click', addNewBmEventHandler);
+				}
+				});
+	
+				lastVid = currentVid;
+		}
+
+	/* respond to messages from background or popup pages */
+	chrome.runtime.onMessage.addListener( (obj,sender, sendResponse) => {
+	
+		if(obj.type=='NEW') 
+			{
+				currentVid = obj.videoId;
+				if (currentVid && lastVid !== currentVid) {
+					newVideoLoaded(false);
+				}
+			}
+		
+		else if ((obj.type=='PLAY')&&(ytPlayer))
+			{
+				ytPlayer.currentTime = obj.value;
+			}
+			
+		else if(obj.type=='DELETE' && BmsList && BmsList.length > 0)
+			{
+				delOldBms();
+				BmsList = BmsList.filter( (b) => ( b.time != obj.value) );
+			}
+		
+		sendResponse({});
+	
+	});
+
+		/* Getting the query parameter for the current tab url on refreshing the tab*/
+		const url_parameters = new URLSearchParams(location.search);
+		currentVid =  url_parameters.get('v');
+	
+		if (currentVid) {
+			newVideoLoaded(true);
+		}
+
+})();
```

Comment:
is there a reason why this needs to be a self executing function??

Reply:
wrap this all in a function(bootup/initialization) and just call it normally.

Reply:
yup, I was using IIFE for variable scoping

Reply:
is this normal for a contentScript?

Reply:
I saw several examples for content scripts using IIFEs, I guess its primary purpose is to prevent polluting the global namespace. hence, it looked like a better way to do it